docs/incremental_replication.md

```diff
@@ -0,0 +1,56 @@
+# Incremental Replication
+
+With incremental replication, a Singer tap emits only data that were created or updated since the previous import rather than the full table.
+
+To support incremental replication, the tap must first define how its replication state will be tracked, e.g. the id of the newest record or the maximal update timestamp in the previous import. Meltano stores the state and makes it available through the context object on subsequent runs. Using the state, the tap should then skip returning rows where the replication key value is less than previous maximal replication key value stored in the state.
+
+## Example Code: Timestamp-Based Incremental Replication
+
+```py
+class CommentsStream(RESTStream):
+
+    replication_key = "date_gmt"
+    is_sorted = True
+
+    schema = th.PropertiesList(
+        th.Property("date_gmt", th.DateTimeType, description="date"),
+    ).to_dict()
+
+    def get_url_params(
+        self, context: dict | None, next_page_token
+    ) -> dict[str, Any]:
+        params = {}
+        if starting_date := self.get_starting_timestamp(context):
+            params["after"] = starting_date.isoformat()
+        if next_page_token is not None:
+            params["page"] = next_page_token
+        self.logger.info(f"QUERY PARAMS: {params}")
+        return params
+
+    url_base = "https://example.com/wp-json/wp/v2/comments"
+    authenticator = None
+```
+
+First we inform the SDK of the `replication_key`, which automatically triggers incremental import mode. Second, optionally, set `is_sorted` to true; with this setting, Singer will throw an error if a supposedly incremental import sends results older than the starting timestamp.
```

Comment:
This is still unclear to me how this property works. Could there be a concrete example provided?

Reply:
I just realized my comment is confusing, what is unclear to me is how the `is_sorted` property is used and I think an example might help.

Like what if the property is set to True and some data points come in, what happens exactly in the messages that are sent to the target? And what if the same data points come in, but the property was set to False, what would be different? Something along those lines.